### DISCUSSION TITLE
Api using puppeteer in production doesn't work

### DISCUSSION THREAD

**Discussion by nunzioono** [vercel/community#1462] [Help] [answered]
### Summary

I've developed a simple app using puppeteer on the api side and displaying the data on the index page, all naturally works in development but after going to production flowlessly and going on the index or directly on the api side the app returns an empty result which is what i expected as result if a try catch block is executed. How can i debug this production issue? Is a known thing or am i ignoring how something works on next js.
The project in question is deployed at: [https://lolesports.vercel.app/](url)

### Additional information

The function returning {} is this one, this is simply called in the handler for api/hello and works in development:

`async function scraper(){
  let pagehtml;
  let response={events:[]};

  try
  {

      const browser = await puppeteer.launch({
          // These flags are need to make puppeteer run in containers.
          args: ["--disable-dev-shm-usage", "--no-sandbox"],
          //product: process.platform!="win32" && process.platform!="darwin"?"firefox":"chrome",
          headless: false
        });
      const page = await browser.newPage();
  
      await page.goto("https://lolesports.com");
      
      await page.waitForSelector('.EventWidgetMatch',{
          visible: true
      })

      await page.content()
      .then(resp=>pagehtml=resp);

      await browser.close();
  
      const $= cheerio.load(pagehtml);

      $('.EventWidgetMatch .match').each(function(_,match) {
          let teams=[];
          let competition;
          let time;

          $(match).find(".team").each(function(_,team){
              let url=$(team).find("img").attr("src");
              let name=$(team).find(".code").html();
              teams.push({name,url})
          });

          competition=$(match).find(".league").text();

          time=$(match).find(".time").text();

          let newmatch={teams,competition,time};
          response.events.push(newmatch);
      });

      return response;
  }
  catch(e)
  {
      return e;
  }
}`

### Example

[https://lolesports.vercel.app/api/hello](url)

**Comment by amyegan** [comment, marked as answer]
Hi, Nunzio! You can get more details in your project's [function logs](https://vercel.com/docs/concepts/deployments/logs#). If you add some logging to your try/catch blocks, that can help you see what is going wrong.
